### PR TITLE
target and crossgen servicehub binaries for .NET 8

### PR DESCRIPTION
**Pull request by genlu**
VS is planning to move to .NET 8 runtime in 17.8, so adjust tfm and crossgen accordingly.
Validation run, RPS result looks good: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/479575

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs [hunk 1]
```diff
@@ -152,7 +152,9 @@ public Assembly LoadAssembly(string assemblyFullName, string? codeBasePath)
                 var assemblyName = new AssemblyName(assemblyFullName);
                 if (!string.IsNullOrEmpty(codeBasePath))
                 {
+#pragma warning disable SYSLIB0044 // Type or member is obsolete
```
Comment (genlu) [inline src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs:155]: @tmat CodeBase is deprecated in .NET 8, any idea what's the right fix here would be?

Comment (tmat) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs:155]: Assembly.LoadFromFile?

Comment (tmat) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs:155]: Or actually, maybe just ignore this? IDK what MEF uses this for.

Comment (genlu) [reply src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspaceManager.cs:155]: hmm, yea, I don't see this method got called. I guess suppressing it is enough

File: eng/Versions.props [hunk 2]
```diff
@@ -327,6 +327,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- We should try to keep this version in sync with the version of app-local runtime in VS. -->
-    <MicrosoftNetCoreAppPackagesVersion>6.0.6</MicrosoftNetCoreAppPackagesVersion>
+    <MicrosoftNetCoreAppRuntimePackagesVersion>8.0.0-preview.5.23280.8</MicrosoftNetCoreAppRuntimePackagesVersion>
+    <MicrosoftWindowsDesktopAppRuntimePackagesVersion>8.0.0-preview.5.23302.2</MicrosoftWindowsDesktopAppRuntimePackagesVersion>
```
Comment (genlu) [inline eng/Versions.props:331]: unlike RTW packages, Preview5 packages for crossgen/runtime and desktopRuntime have different version numbers, therefore I had to split it up.

**Comment by genlu** [comment]
Close this one, we will keep targeting .NET 6 to make sure ti still work with older VS with .NET 6 runtime

**Pull request closed without merging**